public/js/products.js

```diff
@@ -0,0 +1,85 @@
+(function(item, cartCountElement) {
+    [].forEach.call(item, element => {
+      element.addEventListener('click', event => {
+        event.preventDefault();
+        const productId = event.target.id;
+        cartAPI('/cart/add', productId)  
+          .then(res => res.json())
+          .then(function(cart) {
+            cartCountElement.innerHTML = cart.count + ' items';
+          });
+      });
+    });
+  })(
+    document.getElementsByClassName('product-card__buy--btn'),
+    document.getElementsByClassName('nav__end-cart--count')[0]
+  );
+
+  function cartAPI(url, productId) {
+
+    return (
+      fetch(url, {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({ productId })
+      })
+    )
+  }
+
+  (function (item, cartCountElement) {
+    [].forEach.call(item, element => {

```

Comment:
Why you are running these loops to attach event, are you not aware of event delegation?

Reply:
Done